src/com/cft/contactmerge/NameMatchLogic.java

```diff
@@ -0,0 +1,129 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the low level business logic for matching naming properties
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public abstract class NameMatchLogic {
+    /***********************************************************************
+     ****************************** Variables ******************************
+     ***********************************************************************/
+    private final static int hyphenCharLimitYesVsMaybe = 3;
+    private final static String[] namePunctuations = {".", ",", "'", "-"};
```

Comment:
I would split on all punctuation. According to StackOverflow, "Pattern.matches("\\p{Punct}", str)" will do that. Disappointed to see that Java doesn't have a builtin method to do it.

Reply:
Looks like there is a way here -> https://stackoverflow.com/questions/18830813/how-can-i-remove-punctuation-from-input-text-in-java?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

I will try this out